File: CPP/Jaket.cpp
#include "Pakaian.cpp"

// Class Jaket
class Jaket : public Pakaian{
    private:
        // Atribut
        string jenis_penutup;
        bool kupluk;
        int jumlah_saku;

    public:
        // Empty constructor
        Jaket() {}

        // Constructor with parameter
        Jaket(string jenis_penutup, bool kupluk, int jumlah_saku) {
            this->jenis_penutup = jenis_penutup;
            this->kupluk = kupluk;
            this->jumlah_saku = jumlah_saku;
        }

        // Setter and getter for jenis penutup
        void setJenisPenutup(string jenis_penutup) {this->jenis_penutup = jenis_penutup;}
        stirng getJenisPenutup() {return jenis_penutup;}
        
        // Setter and getter for kupluk
        void setKupluk(bool kupluk) {this->kupluk = kupluk;}
        stirng getJenisKupluk() {return kupluk;}
        
        // Setter and getter for jumlah saku
        void setJumlahSaku(int jumlah_saku) {this->jumlah_saku = jumlah_saku;}
        stirng getJumlahSaku() {return jumlah_saku;}

        // Destructor
        ~Jaket() {}
};

File: Java/Jaket.java
// Class turunan Pakaian: data jaket
// Atribut: jenisPenutup, kupluk, jumlahSaku

public class Jaket extends Pakaian {
    private String jenisPenutup;
    private boolean kupluk;
    private int jumlahSaku;

    // Constructor
    public Jaket() {
        super();
        this.jenisPenutup = "";
        this.kupluk = false;
        this.jumlahSaku = 0;
    }

    public Jaket(String idProduk, String merk, float harga,
                 String ukuran, String bahan, String warna,
                 String jenisPenutup, boolean kupluk, int jumlahSaku) {
        super(idProduk, merk, harga, ukuran, bahan, warna);
        this.jenisPenutup = jenisPenutup;
        this.kupluk = kupluk;
        this.jumlahSaku = jumlahSaku;
    }

    // Setter dan getter for jenisPenutup
    public void setJenisPenutup(String jenisPenutup) {
        this.jenisPenutup = jenisPenutup;
    }

    public String getJenisPenutup() {
        return jenisPenutup;
    }

    // Setter dan getter for kupluk
    public void setKupluk(boolean kupluk) {
        this.kupluk = kupluk;
    }

    public boolean isKupluk() {
        return kupluk;
    }

    // Setter dan getter for jumlahSaku
    public void setJumlahSaku(int jumlahSaku) {
        this.jumlahSaku = jumlahSaku;
    }

    public int getJumlahSaku() {
        return jumlahSaku;
    }

    // Destructor: Java memakai garbage collector, method ini hanya
    // disertakan agar sama dengan versi C++
    @Override
    @SuppressWarnings("removal")
    protected void finalize() throws Throwable {
        super.finalize();
    }
}
